tests/unit/client_factory_test.py

```
@@ -0,0 +1,212 @@
+
+class TestClientFromEnv:
+    """Tests for Client.from_env() class method."""
+
+    def test_from_env_missing_operator_id_raises_error(self):
+        """Test that from_env raises ValueError when OPERATOR_ID is missing."""
+        # Generate a real key to prevent validation errors
+        dummy_key = PrivateKey.generate_ed25519().to_string_der()
+        
+        with patch.object(client_module, 'load_dotenv'):
+            with patch.dict(os.environ, {"OPERATOR_KEY": dummy_key}, clear=True):
+                with pytest.raises(ValueError) as exc_info:
```

Ruff (0.14.10)

`pytest.raises(ValueError)` is too broad, set the `match` parameter or use a more specific exception

(PT011)